contracts/SocialController.sol

```diff
@@ -0,0 +1,246 @@
+     */
+    function updateCreatorAddress(address _newAddress, string calldata _username) external {
+        bytes32 hashedUsername = keccak256(abi.encodePacked(_username));
+        uint256 tokenId = idByUsername[hashedUsername];
+        require(tokenId != 0, "This username has not been linked to a creator.");
+
+        Creator memory creator = creatorById[tokenId];
+        require(creator.creatorAddress == msg.sender, "You're not qualified to change this creator's address.");
+        require(creator.creatorAddress != _newAddress, "Your address remains the same.");
+        creatorById[tokenId].creatorAddress = _newAddress;
+
+        emit CreatorAddressUpdated(_username, _newAddress, msg.sender);
+    }
+
+    /**
+     * ! Check that the new username owner is also the owner of the old username
+     * @notice For creator: Update new username after verification on Bloom that the new username belongs
+     * to creator (msg.sender).
+     * @param _newUsername          The new username that replaces the existing one.
+     * @param _oldUsername          The username that the creator wants to change.
+     */
+    function updateUsername(string calldata _newUsername, string calldata _oldUsername) external {
+        bytes32 oldHashedUsername = keccak256(abi.encodePacked(_oldUsername));
+        require(idByUsername[oldHashedUsername] != 0, "Your old username has not been linked.");
+
+        Creator memory creator = creatorById[idByUsername[oldHashedUsername]];
+        // not sure if the following line is necessary as the line after checks for address (as part of Creator struct)
+        require(creator.isCreator, "Your creator profile does not exist.");
+        require(creator.creatorAddress == msg.sender, "You are not allowed to change this username.");
+
+        require(
+            keccak256(abi.encodePacked(_oldUsername)) != keccak256(abi.encodePacked(_newUsername)),
+            "You have not changed your username."
+        );
+        bytes32 newHashedUsername = keccak256(abi.encodePacked(_newUsername));
+        require(idByUsername[newHashedUsername] == 0, "Your new username has already been linked.");
+
+        uint256 tokenId = idByUsername[oldHashedUsername];
+        idByUsername[newHashedUsername] = tokenId;
+        idByUsername[oldHashedUsername] = 0;
+
+        emit CreatorUsernameUpdated(_newUsername, _oldUsername);
+    }
+
+    /**
+     * @notice For users to purchase specific creator tokens by quantity. Tokens are minted by each purchase.
+     * @param _tokenId              The token ID of the creator that the user wishes to buy from.
+     * @param _amount               The amount that the user wishes to buy.
+     */
+    function buyToken(uint256 _tokenId, uint256 _amount) external {
+        require(_amount > 0, "You have not input a valid amount for purchase.");
+
+        Creator memory creator = creatorById[_tokenId];
+        uint256 MAX_SAFE_INTEGER = 9007199254740991;
```

Comment:
- I think the variable name `MAX_SAFE_INTEGER` is kind of confusing.
- I guess, hex(`9007199254740991` ) = '0x1fffffffffffff' is more readable?

Reply:
Updated var name to JS_SAFE_INT and shifted it to become a constant state variable.